src/store.ts

```diff
@@ -163,6 +164,8 @@ export interface Actions {
   projectFlushedToEditor(): void;
 
   setDownload(state: DownloadState): void;
+  // TODO: does the persistence slow this down? we could move it to another store
```

Comment:
Just checking that this is something intended for a subsequent PR?

Reply:
Can also revisit making it more local, but that was hard here as the download can be triggered in different ways with different steps prior to progress. Maybe could look to Python Editor for inspiration.